pkg/codegen/schema/loader.go

```diff
@@ -51,12 +56,66 @@ func (l *pluginLoader) ensurePlugin(pkg string, version *semver.Version) error {
 		Name:    pkg,
 		Version: version,
 	}
+
+	tryDownload := func(dst io.WriteCloser) error {
+		defer dst.Close()
+		tarball, expectedByteCount, err := pkgPlugin.Download()
+		if err != nil {
+			return err
+		}
+		defer tarball.Close()
+		copiedByteCount, err := io.Copy(dst, tarball)
+		if err != nil {
+			return err
+		}
+		if copiedByteCount != expectedByteCount {
+			return fmt.Errorf("Expected %d bytes but copied %d when downloading plugin %s",
+				expectedByteCount, copiedByteCount, pkgPlugin)
+		}
```

Comment:
Curious: have we hit this in practice?

Reply:
Yeah, not sure. Well the CI is hitting something like "connection error" when it's still reading tar contents. Checking byte count sounds like a reasonable thing to introduce, but was not able to repro, so I'm not sure if we ran this, would we hit this check or would we fail before it with "connection error" while copying the bytes, probably the later.